### ISSUE TITLE
Unprocessed tasks disapearing

### ISSUE THREAD

**Issue by at055612** [gchq/stroom#3177]
Raised by another user of stroom

We are having a problem that may be a pretty major one.  We are running v7.0.9.  When we set off a processor job and we monitor it we can see the counts of complete, processing and unprocessed.  The problem is that sometimes the unprocessed is just disappearing without the jobs moving to processing or complete, ie the tasks are being dropped.
Looking at the app.log I am only seeing WARN messages
WARN [<timestamp>} [dw-xxxxx - POST /api/processorTask/v1/assign/node01] stroom.processor.impl.db.ProcessortTaskDaoImpl - changeTaskStatus(ASSIGNED) - Task has changed, attempting reload ProcessorTask(...
WARN [<timestamp>} [dw-xxxxx - POST /api/processorTask/v1/assign/node01] stroom.processor.impl.db.ProcessortTaskDaoImpl - changeTaskStatus(ProcessorTask(...
We are worried that this could be happening quite a lot.  I will be doing some further analysis, is there anything we should increase the log_level for to help diagnose this?

This only seems to be happening to our production environment.  The main difference is that our Prod environment was originally a v4 (possibly earlier) and has had numerous upgrades, whereas our dev envrionment was built at 5.5.13 then upgraded to 7.

Were processing the same data through the same pipeline in both dev and production environment, only seeing the issue in our production environment. 1391 streams being processed, random number of output (i.e not consistent output on multiple runs).

I can also confirm no other system is using the production database.
We have disabled the Processor Task Retention job and this has *fixed* the issue.  I imagine we can run for a while with this job disabled.

I noticed in the app.log that the ProcessorTaskDeleteExecutorImpl was running just prior to the WARN entries which led me to try stopping the Processor Task Retention job.

Again this doesn't seem to be happening in our dev environment.  I've compared the config table and the config.yml files between the environments and nothing stands out.

**Comment by at055612** [comment]
Logging output for the retention job will now look something like
```
INFO  2023-01-13T16:01:29.391Z Stroom P2 #2 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Start 
INFO  2023-01-13T16:01:29.392Z Stroom P2 #2 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Using deleteAge: PT30S, deleteThreshold: 2023-01-13T16:00:59.391988811Z 
INFO  2023-01-13T16:01:29.423Z Stroom P2 #2 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Physically deleted 4 old processor tasks with a state of (COMPLETED|DELETED) 
INFO  2023-01-13T16:01:29.425Z Stroom P2 #2 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Finished in 33ms 

INFO  2023-01-13T16:01:59.392Z Stroom P2 #4 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Start 
INFO  2023-01-13T16:01:59.392Z Stroom P2 #4 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Using deleteAge: PT30S, deleteThreshold: 2023-01-13T16:01:29.392485243Z 
INFO  2023-01-13T16:01:59.415Z Stroom P2 #4 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Logically deleted 1 old processor filters with a state of COMPLETED and no tasks 
INFO  2023-01-13T16:01:59.416Z Stroom P2 #4 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Finished in 24ms 

INFO  2023-01-13T16:02:29.393Z Stroom P2 #3 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Start 
INFO  2023-01-13T16:02:29.393Z Stroom P2 #3 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Using deleteAge: PT30S, deleteThreshold: 2023-01-13T16:01:59.393964532Z 
INFO  2023-01-13T16:02:29.396Z Stroom P2 #3 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - No records logically or physically deleted 
INFO  2023-01-13T16:02:29.396Z Stroom P2 #3 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Finished in 2.0ms 

INFO  2023-01-13T16:02:59.395Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Start 
INFO  2023-01-13T16:02:59.395Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Using deleteAge: PT30S, deleteThreshold: 2023-01-13T16:02:29.395678080Z 
INFO  2023-01-13T16:02:59.428Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Physically deleted 1 old and processor filters with state of DELETED 
INFO  2023-01-13T16:02:59.428Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Finished in 32ms 

INFO  2023-01-13T16:03:39.397Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Start 
INFO  2023-01-13T16:03:39.397Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Using deleteAge: PT30S, deleteThreshold: 2023-01-13T16:03:09.397563179Z 
INFO  2023-01-13T16:03:39.399Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - No records logically or physically deleted 
INFO  2023-01-13T16:03:39.400Z Stroom P2 #7 s.p.i.d.ProcessorTaskDeleteExecutorImpl Processor Task Retention - Finished in 2.0ms 
```

**Comment by at055612** [comment]
Fixed in v7.0.14
